### PR TITLE
make a nicer to_string for Job

### PR DESCRIPTION
**Pull request by BrianHicks**
log output now looks like `d2a0eee0ece469cb (bash -c "echo 'Hello, World!' > out")` instead of a debug struct of the whole job.

Also fixes a configuration bug where we were joining to the workspace directory twice (so making `.rbt/workspaces/workspaces`.)

Based on #34; merge that first!

### CODE REVIEW

File: src/coordinator.rs [hunk 1]
```diff
@@ -17,9 +17,9 @@ pub struct Coordinator {
 }
 
 impl Coordinator {
-    pub fn new(workspace_root: PathBuf, store: Store) -> Self {
+    pub fn new(workspace_root: &Path, store: Store) -> Self {
```
Comment (bhansconnect) [inline src/coordinator.rs:20]: nit general design comment: if you are just gonna take ownership or call `to_path_buf`/`to_owned`/`to_string`/etc, you should take the final type you want as an argument to the function. So in this case, directly take the `PathBuf`. That way, if the caller can give you ownership, no wasted copy is needed. If the caller can't, they can explicitly request the copy. Helps reduce wasted work from accidentally copies if you ultimately want ownership of a value.

File: src/coordinator.rs [hunk 2]
```diff
@@ -64,7 +64,7 @@ impl Coordinator {
                 .store_from_workspace(job, workspace)
                 .context("could not store job output")?;
         } else {
-            log::debug!("already had output of this job; skipping");
+            log::debug!("already had output of job {}; skipping", &job);
```
Comment (bhansconnect) [inline src/coordinator.rs:67]: why does here need `&` but line 55 not?

File: src/fake_runner.rs [hunk 3]
```diff
@@ -8,7 +8,7 @@ pub struct FakeRunner {}
 
 impl Runner for FakeRunner {
     fn run(&self, job: &Job, _workspace: &Workspace) -> Result<()> {
-        log::info!("fake-running job {:?}", job);
+        log::info!("fake-running job {:}", job);
```
Comment (bhansconnect) [inline src/fake_runner.rs:11]: `:` not needed?

**Review by bhansconnect — APPROVED** [review]
All nit comments.

File: src/coordinator.rs (hunk 1 above)
Comment (BrianHicks) [reply src/coordinator.rs:20]: that's super helpful design advice, thank you!

File: src/coordinator.rs (hunk 2 above)
Comment (BrianHicks) [reply src/coordinator.rs:67]: huh, looks like it doesn't actually need it! That's weird; usually Clippy points out unnecessary borrows!

**Pull request merged**